eos-tech-support/eos-download-image

```diff
@@ -56,310 +56,306 @@ else:
+            log("Downloading uncompressed", what, "image signature from", url)
+            full_ext_sig_dir = os.path.dirname(ext_signature)
+            if full_ext_sig_dir:
+                os.makedirs(full_ext_sig_dir, exist_ok=True)
+            self.download(url, ext_signature)
+
+        keyring = self.ensure_keyring()
+
+        log("Verifying")
+        check_call("gpgv", "--keyring", keyring, signature, image)
+
+        log("Download okay.")
+        return image
+
+    def __init__(self):
+        p = argparse.ArgumentParser(
+            description='Fetches and verifies an Endless OS image; '
+                        'fetches the Endless Installer for Windows; '
+                        'or mirrors an entire Endless OS product')
+        p.add_argument('-o', '--outdir',
+                       help='Output directory (default: current directory)',
+                       default='')
+        p.add_argument('-r', '--product',
+                       help='Product (eg: eosinstaller; default: eos)',
+                       default='eos')
+        p.add_argument('-i', '--internal',
+                       action='store_true',
+                       help='Fetch images from the Endless internal network')
+
+        # Mode
+        m = p.add_mutually_exclusive_group()
+        m.add_argument('-m', '--mirror',
+                       action='store_true',
+                       help='Mirror all versions and personalities')
+        m.add_argument('-w', '--windows-tool',
+                       action='store_true',
+                       help='Fetch the Windows USB creator/installer, '
+                            'not an Endless OS image')
+
+        # Which image? NB. product is not part of this group, because it *is*
+        # valid together with --mirror
+        g = p.add_argument_group(title='image selection (ignored with '
+                                       '--mirror or --windows-tool)')
+        g.add_argument('-p', '--personality',
+                       help='Image personality (default: base)',
+                       default='base')
+        g.add_argument('-v', '--version',
+                       help='Image version (default: newest)')
+        g.add_argument('-I', '--iso', action='store_true',
+                       help='Fetch an ISO image, not a raw disk image')
+
+        self.args = args = p.parse_args()
+
+        if args.product == 'fnde' and args.personality == 'base':
+            args.personality = 'fnde_aluno'
+
+        self.base = INTERNAL_BASE_URL if args.internal else PUBLIC_BASE_URL
+
+        # Create the output directory if necessary
+        if args.outdir:
+            os.makedirs(args.outdir, exist_ok=True)
+
+        with requests.sessions.session() as session:
+            self.session = session
+
+            if args.windows_tool:
+                self.fetch_windows_tool()
+            elif args.mirror:
+                self.mirror_images()
+            else:
+                self.fetch_image()
```

Comment:
Yes. I suppose it would be cleaner to have `__main__` go something like:

```python
with requests.session() as session:
    app = EosDownloadImage(session)
    app.run()
```